Software rasterizer

sunverwerth opened the pull request:
Adds a software rasterizer which is still very simplistic:
* Draws every triangle with a red-green-blue gradient
* Has no depth buffer.
* Probably has some precision issues

Also contains functionality to create a context:
`OwnPtr<GLContext> gl_create_context(Gfx::Bitmap& frontbuffer)`

Draft because I still need to clean up the code.

sunverwerth committed:
AK: Add epsilon() to NumericLimits for float, double, long double

sunverwerth committed:
LibGfx: Fix matrix-vector multiplication

sunverwerth committed:
LibGL: Add a software rasterizer

This is still very primitive as it only renders triangles
with a red-green-blue gradient.

Quaker762 commented on Userland/Libraries/LibGL/SoftwareRasterizer.h:
```diff
@@ -0,0 +1,238 @@
+#include <immintrin.h>
+#endif
+
+struct alignas(16) QFloat {
+    union {
+#ifdef USE_SSE
+        __m128 v;
+#endif
+        struct { float a, b, c, d; };
+    };
+
+    constexpr QFloat() = default;
+    constexpr QFloat(float a, float b, float c, float d) :a(a), b(b), c(c), d(d) {}
+#ifdef USE_SSE
+    constexpr QFloat(__m128 v) : v(v) {}
+#endif
+};
```
I believe much of this is supported via `AK/SIMD.h`. According to ccapitalK, "They support all the normal c++ arithmetic operations, plus all the gcc x86 SSE builtins" so the compiler may do this for us if somebody compiles with `-msse`

Quaker762 commented on Userland/Libraries/LibGL/GLContext.cpp:
```diff
@@ -0,0 +1,32 @@
+void gl_make_context_current(GLContext* context)
+{
+    g_gl_context = context;
+}
```
New line here (in case you've missed it)

sunverwerth committed:
LibGL: Add face culling functions

Adds all needed functions to support face culling and implements face
culling in the SoftwareGLContext.
